README.md

```diff
@@ -1,199 +1,201 @@
+  dest: FeedId,
+  value: value, // a weight for the edge source => dest
+}
+```
 
-### get ({source?, dest?}, cb)
+- `opts.old` *Boolean* - whether or not to include the current state (such as
+what `ssb.friends.graph()` returns). (Default: `true`)
+- `opts.live` *Boolean* - whether or not to include subsequent updates of edges
+in the social graph.
+(Default: `false`)
 
-get the follow graph, or a portion of the follow graph. If neither `source` or `dest` are provided,
-the full follow graph will be returned. If `source` is provided, the result will be the map
-of all feeds that the source has relationship edges to (including follow and block), if only
-`dest` is provided, the result will be the same form, but it will represent feeds that follow the `dest`.
-If _both_ `source` and `dest` are provided, the result will be a single value: true if `source` follows `dest`, false if they are blocked, or null if they do not follow (or have unfollowed)
+## Edge weights
 
-### createFriendStream ({meta, live, old}) => Source
+This module is implemented in terms of [dynamic-dijkstra](https://github.com/dominictarr/dynamic-dijkstra)
+(via [layered-graph](https://github.com/ssbc/layered-graph)).
 
-returns a source stream of the hops. if `meta` is true,
-each item is in the form `{id: <id>, hops: <hops>}`, otherwise, each item is just the `<id>`.
-if `old` is false, current values will not be streamed. If live is true, the stream will stay open
-but do nothing until hops changes due to the addition of new edges.
+Relations between feeds are represented as non-zero numbers, as follows:
 
-### stream ({live, old})
+In SSB we use `1` to represent a follow, `-1` to represent a block, `-2` to
+represent unfollow, and `0.1` to represent "same-as".
 
-Create a stream of changes to the graph.
-The first item is the graph structure
-`{<id>: {<id>: <value>,...}, ...}`
+A feed with distance `2` is a "friend of a friend" (we follow someone `+1`
+who follows them `+1` which sums up as `2`). The distance `-2` can mean either
+blocked by a friend or unfollowed by us.
 
-The rest of the values are real time changes, of the form `{from: <id>, to: <id>, value: <value>}`
-value is a boolean or null, as with the other legacy apis.
+If a friend follows someone another friend blocks, the friends follow wins,
+but if you block them directly, that block wins over the friend's follow.
 
-> note: I am considering a graph stream api were the rest are the same form as the graph structure,
-and merging those objects gets the updated graph. for a single edge `<from>-><to>`
-it would look like `{<from>: {<to>: <value>}}`
+"same-as" is represented by very low weights (such as `0.1`). To link two
+devices `a, b` together, we have edges `a->b` and `b->a` with very low weights.
+Low weights can also be used for delegation. Say, a blocklist `L` can be
+implemented as a node that only blocks, then someone `A` subscribes
+to that blocklist by adding edge `A->L` with a weight of `0.1`.
```

Comment:
This text was also in the original README, I didn't write anything new. Not sure how to treat this, I feel like removing it may be a detrimental to people who would like to know more.

Reply:
I think move it to a "future design" section at the end or something. Preface with "this code can sort these ideas, but there's currently no way to activate it" or something?

Reply:
I gave this paragraph some thought, and it sounds technically neat, but I don't think it's something we should build. There's the same-as part, which we agreed to delete, then there is the "blocklist delegation" part, which is like strong-following a peer that only blocks others. It sounds like a hack, would pollute the social graph with "non-person" feeds, and there's definitely other ways of implementing it that are less hacky and more explicit. So I'll just delete that paragraph.